### ISSUE TITLE
nlp_bart_text-error-correction_chinese bart转换到huggingface使用时和modelscope结果有细微的差别

### ISSUE THREAD

**Issue by yayaQAQ** [modelscope/modelscope#65]
## 原因：
modelscope的纠错不支持batch，并且可控性太差，于是想转换至huggingface

### 转换代码：
```
def convert_fairseq_to_huggingface():
    hf_model = torch.load("./huggingface_bart_large/pytorch_model.bin")
    # model2 = torch.load("./bart_hf/pytorch_model.bin")
    fairseq_model = torch.load("./modelscope_bart/pytorch_model.pt")
    model2 = OrderedDict()
    for key, value in fairseq_model['model'].items():
        model2["model."+key] = value
    # model2['final_logits_bias'] = hf_model['final_logits_bias']
    # model2['lm_head.weight'] = hf_model['lm_head.weight']
    torch.save(model2, "./bart_hf/pytorch_model.bin")
```
### 预测代码：
```
def predict_modelscope_correction_via_huggingface():
    tokenizer = BertTokenizer.from_pretrained("./bart_hf/", padding=True,
                                              bos_token="<s>", eos_token="</s>",
                                              pad_token="<pad>", unk_token="<unk>",
                                              cls_token="[CLS]", sep_token="[SEP]",
                                              add_special_tokens=False)
    hf_model_0 = BartForConditionalGeneration.from_pretrained("./bart_hf/")
    inputs = tokenizer.tokenize("一个具有良好内控制度的企业，进行科学的企业管理是十分必要的。")
    inputs += [tokenizer.eos_token]
    input_length = len(inputs)
    inputs = [tokenizer.convert_tokens_to_ids(inputs)]
    inputs = {
        "input_ids": torch.LongTensor(inputs),
        "attention_mask": torch.LongTensor([[1] * input_length])
    }
    summary_ids = hf_model_0.generate(inputs["input_ids"], num_beams=5, output_scores=True, output_attentions=True,
                                      min_length=0, max_length=50, return_dict_in_generate=True, pad_token_id=1,
                                      decoder_start_token_id=2, output_hidden_states=True)
    # print(summary_ids)
    print(tokenizer.batch_decode(summary_ids['sequences'], clean_up_tokenization_spaces=False)[0])
```
我有看modelscope的代码，纠错使用的beam_size=5; decoder_start_id 是 2也就是```</s> ```
然后input_ids 会加入eos_token也就是 ```</s>```
但是使用上面的代码得到的结果和modelscope有较大的差别。
hf:  ```一个具有良好内控制度的企业，进行科学的企業管理是十分必要的。```
ms:```一个具有良好内控制度的企业，进行科学的企业管理是十分必要的。```
但是在示例句子上结果是一致的。即：```这洋的话，下一年的福气来到自己身上。```

**Comment by yayaQAQ** [comment]
### 如果其他人也要使用的话，还需要加入config.json以及vocab.txt 如下：
#### config.json
```
{
  "activation_dropout": 0.1,
  "activation_function": "gelu",
  "add_bias_logits": false,
  "add_final_layer_norm": false,
  "architectures": [
    "BartForConditionalGeneration"
  ],
  "attention_dropout": 0.1,
  "bos_token_id": 0,
  "classif_dropout": 0.1,
  "classifier_dropout": 0.0,
  "d_model": 1024,
  "decoder_attention_heads": 16,
  "decoder_ffn_dim": 4096,
  "decoder_layerdrop": 0.0,
  "decoder_layers": 12,
  "decoder_start_token_id": 0,
  "dropout": 0.1,
  "early_stopping": true,
  "encoder_attention_heads": 16,
  "encoder_ffn_dim": 4096,
  "encoder_layerdrop": 0.0,
  "encoder_layers": 12,
  "eos_token_id": 2,
  "forced_eos_token_id": 2,
  "gradient_checkpointing": false,
  "id2label": {
    "0": "LABEL_0",
    "1": "LABEL_1",
    "2": "LABEL_2"
  },
  "init_std": 0.02,
  "is_encoder_decoder": true,
  "label2id": {
    "LABEL_0": 0,
    "LABEL_1": 1,
    "LABEL_2": 2
  },
  "max_position_embeddings": 512,
  "model_type": "bart",
  "no_repeat_ngram_size": 3,
  "normalize_before": false,
  "normalize_embedding": true,
  "num_beams": 4,
  "num_hidden_layers": 12,
  "pad_token_id": 0,
  "scale_embedding": false,
  "task_specific_params": {
    "summarization": {
      "length_penalty": 1.0,
      "max_length": 128,
      "min_length": 12,
      "num_beams": 4
    },
    "summarization_cnn": {
      "length_penalty": 2.0,
      "max_length": 142,
      "min_length": 56,
      "num_beams": 4
    },
    "summarization_xsum": {
      "length_penalty": 1.0,
      "max_length": 62,
      "min_length": 11,
      "num_beams": 6
    }
  },
  "transformers_version": "4.4.1",
  "use_cache": true,
  "tokenizer_class": "BertTokenizer",
  "vocab_size": 21132
}
```
[vocab.txt](https://github.com/modelscope/modelscope/files/10358077/vocab.txt)

**Comment by yayaQAQ** [comment]
"no_repeat_ngram_size": 3,
将这个参数从config.txt当中移除，效果就正常了。

**Comment by yayaQAQ** [comment]
{
  "activation_function": "gelu",
  "add_bias_logits": false,
  "add_final_layer_norm": false,
  "architectures": [
    "BartForConditionalGeneration"
  ],
  "bos_token_id": 0,
  "classifier_dropout": 0.0,
  "d_model": 1024,
  "decoder_attention_heads": 16,
  "decoder_ffn_dim": 4096,
  "decoder_layerdrop": 0.0,
  "decoder_layers": 12,
  "decoder_start_token_id": 0,
  "early_stopping": true,
  "encoder_attention_heads": 16,
  "encoder_ffn_dim": 4096,
  "encoder_layerdrop": 0.0,
  "encoder_layers": 12,
  "eos_token_id": 2,
  "forced_eos_token_id": 2,
  "gradient_checkpointing": false,
  "is_encoder_decoder": true,
  "max_position_embeddings": 512,
  "model_type": "bart",
  "normalize_before": false,
  "normalize_embedding": true,
  "num_beams": 4,
  "num_hidden_layers": 12,
  "pad_token_id": 0,
  "scale_embedding": false,
  "transformers_version": "4.4.1",
  "use_cache": true,
  "tokenizer_class": "BertTokenizer",
  "vocab_size": 21132
}

附上修改后的config.json
